.github/workflows/main.yml

```diff
@@ -1,6 +1,11 @@
 name: main
 
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+  schedule:
+    # Prime the caches every Monday
+    - cron: 0 1 * * MON
```

Comment:
Hope I'm reading this correctly: runs the CI every Monday at 1 AM. Trying to understand why we'd need to run this cron job on top of the PR and push events.

Reply:
It's an important point. The caching mechanism used in the GitHub Action is that if the cache is not used, the cache will expire in two weeks and you will need a clean build. So this allows you to extend the cache lifetime every week.

Reply:
I see, that makes sense. Thanks for the explanation.